src/Routing/Middleware/RoutingMiddleware.php

```diff
@@ -60,11 +73,45 @@ protected function loadRoutes()
         if (!$this->app) {
             return;
         }
+
+        $routeCollection = $this->buildRouteCollection();
+        Router::setRouteCollection($routeCollection);
+    }
+
+    /**
+     * Check if route cache is enabled and use the configured Cache to 'remember' the route collection
+     *
+     * @return \Cake\Routing\RouteCollection
+     */
+    protected function buildRouteCollection()
+    {
+        $isRouterCacheEnabled = Configure::read('Router.cache');
```

Comment:
Passing in the cache engine name would remove the need to have a `Router.cache` configure option as well which is nice.

Reply:
The intention here was to provide a way to override this behavior locally, for example in development vs production, and allow an easy environment based configuration.

I can refactor the middleware constructor to pass the cache configuration name (or null to disable) then set defaults in Application and remove the need of these 2 config keys if you think it would be better to reduce the amount of config keys.

Thanks,

Reply:
The caller can manage configuration keys too. Doing that makes the middleware easier to adapt to di containers in the future as well.

Reply:
true, OK I'll do these changes and remove config from the middleware and read the config from the Application class